cl/search/models.py

```diff
@@ -871,6 +877,8 @@ def pacer_docket_url(self):
 
             if not self.pacer_case_id:
                 return self.pacer_appellate_url_with_caseNum(path)
+            elif self.pacer_case_id.count("-") == 4:
+                return self.pacer_acms_url()
```

Comment:
Is this really wise? To decide we are following ACMS URL rules because there are exactly four hyphens in the PACER case ID, i.e. it's presumably a GUID? This just feels fragile and like we're overloading fields.

Is the real problem that AbstractPacerDocument lacks a type field? Should we just open an Issue to address that (later)? idk.

Reply:
> Is the real problem that AbstractPacerDocument lacks a type field?

A type field would be really helpful, it will provide a clear and concise identifier for ACMS docket. Currently, the only means of differentiation is the "pacer_case_id".

> This just feels fragile.

You're right. I think can use the [uuid](https://docs.python.org/3/library/uuid.html) module to check whether the `pacer_doc_id` is a valid GUID or not:

``` python
def is_valid_uuid(uuid_to_test):
    try:
        uuid_obj = UUID(uuid_to_test)
        return True
    except ValueError:
        return False
```

and then:


```python
elif is_valid_uuid(self.pacer_case_id):
      return self.pacer_acms_url()
```

Reply:
> You're right. I think can use the [uuid](https://docs.python.org/3/library/uuid.html) module to check whether the `pacer_doc_id` is a valid GUID or not:

We could, but that's arguably even worse. We don't enforce that requirement anywhere else (deliberately so, in my view!) Look at the tests, e.g. `PacerFetchQueueSerializer` that checks if `attrs.get("pacer_case_id").count("-")` is not equal to `1`. 

If we allow an invalid GUID to be inserted (which we probably should. If `ca9` gives us a docket id of `correct-horse-battery-staple`, who are we to say they are wrong?), and we don't validate against the uuid spec anywhere else, we should not do so here.

Also, according to https://stackoverflow.com/questions/53847404/how-to-check-uuid-validity-in-python#comment108940714_33245493, your test is insufficient, a `ValueError` is not always raised.
It's that kind of subtlety and complexity that argues against that kind of check.